core/controllers/resources.py

```diff
@@ -86,3 +87,25 @@ def get(self, exploration_id, asset_type, encoded_filename):
             self.response.write(raw)
         except:
             raise self.PageNotFoundException
+
+
+class PromoBarHandler(base.BaseHandler):
+    """The handler for checking if Promobar is enabled and fetching
+    promobar message.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    REDIRECT_UNFINISHED_SIGNUPS = False
```

Comment:
Add a comment here explaining why, for future devs.

Reply:
Done.

Reply:
Hi, @abhaygupta97 Glad that you figured out the solution to the 'session expired' problem! And also thanks for the detailed flowchart! :smile: